src/rust/src/x509/common.rs

```diff
@@ -646,6 +647,52 @@ pub(crate) fn encode_extensions<
     )))
 }
 
+#[pyo3::prelude::pyfunction]
+fn encode_extension_value<'p>(
+    py: pyo3::Python<'p>,
+    py_ext: &'p pyo3::PyAny,
+) -> pyo3::PyResult<&'p pyo3::types::PyBytes> {
+    let unrecognized_extension_type: &pyo3::types::PyType = py
+        .import("cryptography.x509")?
+        .getattr("UnrecognizedExtension")?
+        .extract()?;
+    if unrecognized_extension_type.is_instance(py_ext)? {
+        return Ok(pyo3::types::PyBytes::new(
+            py,
+            py_ext.getattr("value")?.extract::<&[u8]>()?,
+        ));
+    }
+
+    let oid = asn1::ObjectIdentifier::from_string(
+        py_ext
+            .getattr("oid")?
+            .getattr("dotted_string")?
+            .extract::<&str>()?,
+    )
+    .unwrap();
+
+    for encode_ext in [
```

Comment:
I don't really like this approach but I'm curious what @alex thinks. I would prefer a new func that just goes through all extensions. However, you'd need to then have tests for serializing every ext to ensure coverage (which isn't necessarily a bad thing).

Reply:
ping @alex

Reply:
If we got rid of the thing where you can only serialize some extensions with some structures, then this would be much easier. Any objections to doing that?

Reply:
So allow users to serialize whatever, even if it isn't valid for the structure based on our reading of RFC 5280? I have mixed feelings about that, but it is still syntactically valid ASN.1.

Reply:
Does 5280 say you can't put a SAN in an OCSP response? I just thought that was us being practical.

Reply:
5280 has no opinions about OCSP at all 😄 But 6960 says under section 4:

```
This section defines some standard extensions, based on the extension
   model employed in X.509 version 3 certificates (see [RFC5280]).
   Support for all extensions is optional for both clients and
   responders.  For each extension, the definition indicates its syntax,
   processing performed by the OCSP responder, and any extensions that
   are included in the corresponding response.
```

CSR, OCSP, and Cert/CRL are all in different RFCs, but in 5280 (where cert and CRL are) there are distinct sections for cert and CRL extensions.

All that said, since arbitrary extensions are part of the spec an extension that just so happens to be exactly the same as one defined for another type is valid so it's not a big stretch to say we should just collapse it all into one big parser.

Reply:
Ok, so how should we proceed with this? Is this something you want to implement? Or should I implement that (in this PR, in a separate PR, ...)?

Reply:
Let's do this as two PRs:
* Unify the serialization into one function and call it from all the serialization methods.
* Use that method here

Reply:
And if you're willing to supply that first PR that'd be great!

Reply:
I can definitely give it a try!